File: public/sdk/python/adsgalaxy.py
import requests


class AdsGalaxyClient:
    def __init__(self, api_key: str, base_url: str = "https://app.adsgalaxy.online"):
        if not api_key:
            raise ValueError("AdsGalaxy api_key is required")
        self.api_key = api_key
        self.base_url = base_url.rstrip("/")

    def request(self, path: str, payload: dict | None = None) -> dict:
        response = requests.post(
            f"{self.base_url}{path}",
            json=payload or {},
            headers={"Content-Type": "application/json", "x-api-key": self.api_key},
            timeout=15,
        )
        data = response.json() if response.content else {}
        if not response.ok:
            raise RuntimeError(data.get("error", "AdsGalaxy API request failed"))
        return data

    def request_rewarded_ad(self, payload: dict) -> dict:
        return self.request("/api/v1/rewarded/request", payload)

    def verify_reward(self, payload: dict) -> dict:
        return self.request("/api/v1/rewarded/verify", payload)

    def postback(self, payload: dict) -> dict:
        return self.request("/api/v1/postbacks", payload)

    def load_campaign(self, payload: dict) -> dict:
        return self.request("/api/v1/bot/campaigns", payload)

    def track_conversion(self, payload: dict) -> dict:
        return self.request("/api/v1/bot/events", {"event_type": "conversion", **payload})

File: public/sdk/javascript/adsgalaxy.js
export class AdsGalaxyClient {
  constructor({ apiKey, baseUrl = "https://app.adsgalaxy.online", metadata = {} }) {
    if (!apiKey) throw new Error("AdsGalaxy apiKey is required");
    this.apiKey = apiKey;
    this.baseUrl = baseUrl.replace(/\/$/, "");
    this.metadata = metadata;
  }

  async request(path, body = {}) {
    const response = await fetch(`${this.baseUrl}${path}`, {
      method: "POST",
      headers: { "Content-Type": "application/json", "x-api-key": this.apiKey },
      body: JSON.stringify({ metadata: this.metadata, ...body }),
    });
    const data = await response.json().catch(() => ({}));
    if (!response.ok) throw new Error(data.error || "AdsGalaxy API request failed");
    return data;
  }

  loadAd(payload) { return this.request("/api/v1/ads/request", payload); }
  requestAd(payload) { return this.loadAd(payload); }
  trackImpression(requestId, payload = {}) { return this.request("/api/v1/ads/events", { request_id: requestId, event_type: "impression", ...payload }); }
  trackClick(requestId, payload = {}) { return this.request("/api/v1/ads/events", { request_id: requestId, event_type: "click", ...payload }); }
  trackCompletion(requestId, payload = {}) { return this.request("/api/v1/ads/events", { request_id: requestId, event_type: "completion", ...payload }); }
  trackReward(requestId, payload = {}) { return this.request("/api/v1/ads/events", { request_id: requestId, event_type: "reward", ...payload }); }
  verifyReward(payload) { return this.request("/api/v1/rewarded/verify", payload); }
  requestRewardedAd(payload) { return this.request("/api/v1/rewarded/request", payload); }
  rewardCallback(payload) { return this.request("/api/v1/rewarded/callback", payload); }
  postback(payload) { return this.request("/api/v1/postbacks", payload); }
}
